app/models/manageiq/providers/amazon/manager_mixin.rb

```diff
@@ -71,7 +71,10 @@ def verify_credentials(args)
       access_key, secret_access_key, proxy_uri, service_account = default_endpoint&.values_at(
         "userid", "password", "proxy_uri", "service_account"
       )
+
       secret_access_key = MiqPassword.try_decrypt(secret_access_key)
+      # Pull out the password from the database if a provider ID is available
+      secret_access_key ||= find(args["id"]).authentications.find_by(:authtype => 'default').password
```

Comment:
`.authentications.find_by(:authtype => 'default').password` is just `.authentication_password('default')`
Can we pass in authtype from the params in case there is more than one auth passed in?

Reply:
Sure, it even looks better :+1: